src/coreclr/jit/codegenwasm.cpp

```diff
@@ -204,6 +528,162 @@ void CodeGen::genCodeForLclVar(GenTreeLclVar* tree)
     }
 }
 
+//------------------------------------------------------------------------
+// genCodeForCompare: Produce code for a GT_EQ/GT_NE/GT_LT/GT_LE/GT_GE/GT_GT node.
+//
+// Arguments:
+//    tree - the node
+//
+void CodeGen::genCodeForCompare(GenTreeOp* tree)
+{
+    assert(tree->OperIsCmpCompare());
+
+    GenTree* const  op1     = tree->gtOp1;
+    var_types const op1Type = op1->TypeGet();
+
+    if (varTypeIsFloating(op1Type))
```

Comment:
You may consider folding the two of these into a single function; there is no real difference between the two types of compares on WASM, so this factoring looks a bit artificial. Though it is consistent with other targets.